demo/site/src/util/graphQLClient.ts

```diff
@@ -0,0 +1,10 @@
+import { createFetchWithDefaults, createFetchWithPreviewHeaders, createGraphQLFetch as createGraphQLFetchLibrary } from "@comet/cms-site";
+import { SitePreviewData } from "@src/app/api/site-preview/route";
+
+export const graphQLApiUrl = `${typeof window === "undefined" ? process.env.API_URL_INTERNAL : process.env.NEXT_PUBLIC_API_URL}/graphql`;
+export function createGraphQLFetch(previewData?: SitePreviewData) {
+    return createGraphQLFetchLibrary(
+        createFetchWithDefaults(createFetchWithPreviewHeaders(fetch, previewData), { next: { revalidate: 15 * 60 } }),
```

Comment:
Do we really need three functions for this? Couldn't this be something that `createGraphQLFetchLibrary` does?

Reply:
I like the idea of flexible composition of different, independent features. I have also the client side cache for https://github.com/vivid-planet/comet/pull/1314 in mind that will use similar patterns